pkg/apis/nmstate/v1/nodenetworkstate_types.go

```diff
@@ -32,12 +33,8 @@ type NodeNetworkState struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   NodeNetworkStateSpec   `json:"spec,omitempty"`
-	Status NodeNetworkStateStatus `json:"status,omitempty"`
-}
-
-type State interface {
-	DeepCopyState() State
+	Spec         NodeNetworkStateSpec `json:"spec,omitempty"`
+	CurrentState State                `json:"currentState,omitempty"`
```

Comment:
We have to get the unit test PR merged first.

Reply:
Rebased in the unit test PR